File: go_algorithms/algorithms/RowByRowSequential.go
package algorithms

import (
	"fmt"
	"time"
)

func RowByRowSequential(B, C [][]int) ([][]int, int64) {
	fmt.Println("Método: RowByRowSequential")

	startTime := time.Now().UnixNano()
	size := len(C)
	A := make([][]int, size)
	for i := range A {
		A[i] = make([]int, size)
	}

	for i := 0; i < size; i++ {
		for j := 0; j < size; j++ {
			for k := 0; k < size; k++ {
				A[i][k] += B[i][j] * C[j][k]
			}
		}
	}

	endTime := time.Now().UnixNano()
	executionTime := endTime - startTime

	return A, executionTime
}

File: python_algorithms/algorithms/RowByRowSequential.py
import numpy as np
import time

def Multiply(B, C):
    print("Método: RowByRowSequential")

    start_time = time.time_ns()

    size = len(C)
    A = np.zeros((size, size), dtype=np.int64)

    for i in range(size):
        for j in range(size):
            for k in range(size):
                A[i][k] += B[i][j] * C[j][k]

    end_time = time.time_ns()
    execution_time = end_time - start_time

    return A, execution_time
